rubrik_cdm/data_management.py

```diff
@@ -1564,34 +1564,46 @@ def _validate_sql_recovery_point(self, mssql_id, date, time, timeout=30):  # pyl
         """
-
-        self.log("_validate_sql_recovery_point: Getting the recoverable range for mssql db ID:'{}'.".format(mssql_id))
-        range_summary = self.get('v1', '/mssql/db/{}/recoverable_range'.format(mssql_id), timeout=timeout)
-
-        self.log("_validate_sql_recovery_point: Converting the provided date/time into UTC.")
-        recovery_date_time = self._date_time_conversion(date, time)
-        recovery_date_time = datetime.strptime(recovery_date_time, '%Y-%m-%dT%H:%M')
-        recovery_timestamp = int(recovery_date_time.strftime('%s')) * 1000
-
-        for range in range_summary['data']:
-            startstr = range['beginTime']
-            endstr = range['endTime']
-            startsplit = startstr[:16]
-            endsplit = endstr[:16]
-            start = datetime.strptime(startsplit, '%Y-%m-%dT%H:%M')
-            end = datetime.strptime(endsplit, '%Y-%m-%dT%H:%M')
-
-            self.log("_validate_sql_recovery_point: Searching for the provided recovery_point.")
-            is_recovery_point = self._time_in_range(start, end, recovery_date_time)
-            if is_recovery_point == False:
-                continue
-            else:
-                break
-        recovery_point = {}
-        recovery_point['is_recovery_point'] = is_recovery_point
-        recovery_point['recovery_timestamp'] = recovery_timestamp
-
-        return recovery_point
+        is_recovery_point = False
+        if date and time == 'latest':
+            latest_data = self.get('v1', '/mssql/db/{}/snapshot'.format(mssql_id), timeout=timeout)
+            try:
+                latest_date_time = latest_data['data'][0]['date']
+            except:
+                raise InvalidParameterException(
+                    "The database with ID {} does not have any existing snapshots.".format(mssql_id))
+
+            data_str = datetime.strptime(latest_date_time[:16], '%Y-%m-%dT%H:%M')
+            date_str, time_str = [data_str.strftime('%m-%d-%Y'), data_str.strftime('%I:%M %p')]
+            recovery_date_time = self._date_time_conversion(date_str, time_str)
+            recovery_date_time = datetime.strptime(recovery_date_time, '%Y-%m-%dT%H:%M')
+            recovery_timestamp = int(recovery_date_time.strftime('%s')) * 1000
+            is_recovery_point = True
+        else:
+            self.log("_validate_sql_recovery_point: Getting the recoverable range for mssql db ID:'{}'.".format(mssql_id))
+            range_summary = self.get('v1', '/mssql/db/{}/recoverable_range'.format(mssql_id), timeout=timeout)
+
+            self.log("_validate_sql_recovery_point: Converting the provided date/time into UTC.")
+            recovery_date_time = self._date_time_conversion(date, time)
+            recovery_date_time = datetime.strptime(recovery_date_time, '%Y-%m-%dT%H:%M')
+            recovery_timestamp = int(recovery_date_time.strftime('%s')) * 1000
```

Comment:
Let's add some comments here to explain whats going on during each step. Date/time stuff can be confusiong.

Reply:
I agree, had to read through a couple of times 👍